project/Component/nef/JekyllAPI.swift

```diff
@@ -32,43 +31,57 @@ public extension JekyllAPI {
+                        .page(content: content, permalink: permalink, filename: filename, into: output)
+                        .contramap(environment)
+                        .mapError { _ in nef.Error.jekyll }
     }
     
-    static func render(playground: URL, into output: URL) -> EnvIO<Console, nef.Error, [URL]> {
-        #warning("It must be completed in NefJekyll module refactor")
-        fatalError()
-//        NefJekyll.Jekyll()
-//                 .renderPlayground(playground, into: output)
-//                 .contramap(environment)
-//                 .mapError { _ in nef.Error.jekyll }^
+    static func render(playground: URL, into output: URL) -> EnvIO<Console, nef.Error, NEA<URL>> {
+        NefJekyll.Jekyll()
+                 .playground(playground, into: output)
+                 .contramap(environment)
+                 .mapError { _ in nef.Error.markdown }^
     }
     
-    static func render(playgroundsAt: URL, mainPage: URL, into output: URL) -> EnvIO<Console, nef.Error, [URL]> {
-        #warning("It must be completed in NefJekyll module refactor")
-        fatalError()
-//        NefJekyll.Jekyll()
-//                 .renderPlaygrounds(at: playgroundsAt, mainPage: mainPage, into: output)
-//                 .contramap(environment)
-//                 .mapError { _ in nef.Error.jekyll }^
+    static func render(playgroundsAt: URL, mainPage: URL, into output: URL) -> EnvIO<Console, nef.Error, NEA<URL>> {
+        NefJekyll.Jekyll()
+                 .playgrounds(at: playgroundsAt, mainPage: mainPage, into: output)
+                 .contramap(environment)
+                 .mapError { _ in nef.Error.jekyll }^
+    }
+    
+    // MARK: api <IO>
+    static func renderIO(content: String, permalink: String) -> IO<nef.Error, String> {
+        render(content: content, permalink: permalink).provide(MacDummyConsole())
+    }
+    
+    static func renderVerboseIO(content: String, permalink: String) -> IO<nef.Error, (ast: String, rendered: String)> {
+        renderVerbose(content: content, permalink: permalink).provide(MacDummyConsole())
+    }
+    
+    static func renderIO(content: String, permalink: String, toFile file: URL) -> IO<nef.Error, URL> {
+        render(content: content, permalink: permalink, toFile: file).provide(MacDummyConsole())
+    }
+    
+    static func renderVerboseIO(content: String, permalink: String, toFile file: URL) -> IO<nef.Error, (url: URL, ast: String, rendered: String)> {
+        renderVerbose(content: content, permalink: permalink, toFile: file).provide(MacDummyConsole())
+    }
+    
+    static func renderIO(playground: URL, into output: URL) -> IO<nef.Error, NEA<URL>> {
+        render(playground: playground, into: output).provide(MacDummyConsole())
+    }
+    
+    static func renderIO(playgroundsAt: URL, mainPage: URL, into output: URL) -> IO<nef.Error, NEA<URL>> {
+        render(playgroundsAt: playgroundsAt, mainPage: mainPage, into: output).provide(MacDummyConsole())
```

Comment:
This should be in the protocol declaration as extension methods.

Reply:
offline-conversation: we only will use the `EnvIO` API and will add support to IO (provide dummy console in `unsafeRun` with some utils)